From f8ad90a0d76741b14767e1e94c9fbe6d760e301d Mon Sep 17 00:00:00 2001
From: Adam Williamson <awilliam@redhat.com>
Date: Tue, 26 Jun 2018 12:14:56 -0700
Subject: [PATCH 2/3] DNF 3: Update size calculations for transaction item
 changes

Seems we can basically get sizes for the package that backs a
transaction item now, rather than for the 'set of files it
installs'.

Signed-off-by: Adam Williamson <awilliam@redhat.com>
---
 pyanaconda/payload/dnfpayload.py | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/pyanaconda/payload/dnfpayload.py b/pyanaconda/payload/dnfpayload.py
index 8b8be68c9..27bc6d7d3 100644
--- a/pyanaconda/payload/dnfpayload.py
+++ b/pyanaconda/payload/dnfpayload.py
@@ -602,7 +602,7 @@ class DNFPayload(payload.PackagePayload):
         if transaction is None:
             return Size(0)
 
-        size = sum(tsi.installed.downloadsize for tsi in transaction)
+        size = sum(tsi.pkg.downloadsize for tsi in transaction)
         # reserve extra
         return Size(size) + Size("150 MB")
 
@@ -782,9 +782,9 @@ class DNFPayload(payload.PackagePayload):
         files_nm = 0
         for tsi in transaction:
             # space taken by all files installed by the packages
-            size += tsi.installed.installsize
+            size += tsi.pkg.installsize
             # number of files installed on the system
-            files_nm += len(tsi.installed.files)
+            files_nm += len(tsi.pkg.files)
 
         # append bonus size depending on number of files
         bonus_size = files_nm * BONUS_SIZE_ON_FILE
-- 
2.18.0.rc2

